snuba/admin/static/runtime_config/index.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from "react";
+import { Table } from "../table";
+import Client from "../api_client";
+import { ConfigKey, ConfigValue, ConfigType, RowData } from "./types";
+
+import { containerStyle, paragraphStyle } from "./styles";
+
+function RuntimeConfig(props: { api: Client }) {
+  const { api } = props;
+
+  // Data from the API
+  const [data, setData] = useState<
+    { key: ConfigKey; value: ConfigValue; type: ConfigType }[] | null
+  >(null);
+
+  // Load data if it was not previously loaded
+  if (data === null) {
+    fetchData();
+  }
+
+  function fetchData() {
+    api.getConfigs().then((res) => {
+      setData(res);
+    });
+  }
```

Comment:
I don't think it really matters for this but just for my education:

Does this make it so that the render is blocked until data is fetched because this is a synchronous function

Reply:
We don't actually block rendering of the component. Basically we are just rendering an empty component until some data is received from the API then we show the table. React automatically triggers re-render soon after the state is updated by the `setData` hook, but it happens asynchronously.